### DISCUSSION TITLE
Unable to get IAM security credentials from EC2 Instance Metadata Service.

### DISCUSSION THREAD

**Discussion by nenne70** [aws/aws-lambda-dotnet#1649] [Q&A] [answered]
Hi

Not sure what has happened. But suddenly The Mock Lambda Test Tool is unable to get the credentials. I am using SSO and have two profiles. 

Both profiles are created using aws configure. I have also tried creating a profile from visual studio. But that also fails with same error. 

I am able to select a profile in the Mock Lambda Test Tool, but it seems to have no effect at all. If I set a default profile that is not using SSO, and only uses long term credentials everything is working.

Running Visual Studio 2022 (17.8.4 same issue with 17.8.3 not sure about earlier)
AWS Toolkit: 1.46.0.0

Any assistance would be appreciated, don't want to use long term credentials, since I am working with multiple aws accounts.

**Comment by ashishdhingra** [comment, marked as answer]
@nenne70 Have you executed `aws sso login` via AWS CLI before launching Lambda Test Tool? Also, what is the error/exception you are getting in the Lambda test tool output?
